File: julia/grains/grains.jl
"""Calculate the number of grains on square `square`."""
function on_square(square)
    if square < 1 || square > 64
        throw(DomainError(""))
    end
    return 2^(square-1) 
end

"""Calculate the total number of grains after square `square`."""
function total_after(square)
    if square < 1 || square > 64
        throw(DomainError(""))
    end
    sum = 0
    for i in 1:square
        sum = sum + on_square(square)
    end
    return sum
end

File: go/grains/grains.go
package grains

import (
	"errors"
	"math"
)

func Square(number int) (uint64, error) {
	if number < 1 || number > 64 {
		return 0, errors.New("Invalid value")
	}
	return uint64(math.Pow(2, float64(number-1))), nil
}

func Total() uint64 {
	var total uint64
	total = 0
	for i := 1; i < 65; i++ {
		grainsOnSquare, _ := Square(i)
		total = total + grainsOnSquare
	}
	return uint64(total)
}
